### ISSUE TITLE
Private API Gateway DNS Issue

### ISSUE THREAD

**Issue by john-odonnell-accenture** [hashicorp/terraform-provider-aws#25640]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform v0.15.3 (also tested v1.2.4)
on windows_amd64
+ provider registry.terraform.io/hashicorp/aws v4.20.1
+ provider registry.terraform.io/hashicorp/random v3.3.2

### Affected Resource(s)

aws_api_gateway_rest_api

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
data "aws_region" "current" {}

resource "aws_default_security_group" "test" {
  vpc_id = data.aws_vpc.vpc.id
}

resource "aws_vpc_endpoint" "test" {
  private_dns_enabled = false
  security_group_ids  = [aws_default_security_group.test.id]
  service_name        = "com.amazonaws.${data.aws_region.current.name}.execute-api"
  subnet_ids          = [
      data.aws_subnet.private-subnet-a.id,
      data.aws_subnet.private-subnet-b.id,
      data.aws_subnet.private-subnet-c.id
    ]
  vpc_endpoint_type   = "Interface"
  vpc_id              = data.aws_vpc.vpc.id
}

resource "aws_api_gateway_rest_api" "test" {
  name = "test"

  endpoint_configuration {
    types            = ["PRIVATE"]
    vpc_endpoint_ids = [aws_vpc_endpoint.test.id]
  }

  body = jsonencode({
    swagger = "2.0"
    info = {
      title   = "test"
      version = "2017-04-20T04:08:08Z"
    }
    schemes = ["https"]
    paths = {
      "/test" = {
        get = {
          responses = {
            "200" = {
              description = "OK"
            }
          }
          x-amazon-apigateway-integration = {
            httpMethod = "GET"
            type       = "HTTP"
            responses = {
              default = {
                statusCode = 200
              }
            }
            uri = "https://api.example.com/"
          }
        }
      }
    }
    x-amazon-apigateway-endpoint-configuration = {
      vpcEndpointIds = [aws_vpc_endpoint.test.id]
    }
  })
}

resource "random_string" "random" {
    length = 4
    special = false
    }

resource "aws_api_gateway_deployment" "api_gateway_deployment-test-private" {
  rest_api_id = aws_api_gateway_rest_api.test.id
  depends_on  = [
    aws_api_gateway_rest_api.test,
    aws_api_gateway_rest_api_policy.test,
    ]

   triggers = {
    redeployment = random_string.random.result 
   }

  lifecycle {
    create_before_destroy = true
  }
}


resource "aws_api_gateway_rest_api_policy" "test" {
  rest_api_id = aws_api_gateway_rest_api.test.id

  policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "AWS": "*"
      },
      "Action": "execute-api:Invoke",
      "Resource": "${aws_api_gateway_rest_api.test.execution_arn}",
      "Condition": {
        "IpAddress": {
          "aws:SourceIp": "10.33.52.0/24"
        }
      }
    }
  ]
}
EOF
}

data "aws_vpc" "vpc" {
  tags = {
    Name = local.vpc-name 
  }
}

data "aws_subnet" "private-subnet-a" {
  filter {
    name   = "vpc-id"
    values = [data.aws_vpc.vpc.id]
  }
  tags = {
   Name = "${local.subnet-name-prefix}-pri-a"
  }
}

data "aws_subnet" "private-subnet-b" {
  filter {
    name   = "vpc-id"
    values = [data.aws_vpc.vpc.id]
  }
  tags = {
   Name = "${local.subnet-name-prefix}-pri-b"
  }
}

data "aws_subnet" "private-subnet-c" {
  filter {
    name   = "vpc-id"
    values = [data.aws_vpc.vpc.id]
  }
  tags = {
   Name = "${local.subnet-name-prefix}-pri-c"
  }
}

locals{
vpc-name           = "test-dev-vpc"
subnet-name-prefix = "test-dev-net"
}

output "api-dns" {
      value = "https://${aws_api_gateway_rest_api.test.id}-${aws_vpc_endpoint.test.id}.execute-api.${data.aws_region.current.name}.amazonaws.com"
  description = "api-dns"
}

provider "aws" {
region = "us-west-2"
}

```

### Expected Behavior

Expecting to get dns response for "https://c6lp4eoh2d-vpce-0c5db4e8da3c23739.execute-api.us-east-1.amazonaws.com"

### Actual Behavior

DNS record is never registered for the private API Gateway endpoint

Following Terraform Apply:
C:\Users>nslookup https://c6lp4eoh2d-vpce-0c5db4e8da3c23739.execute-api.us-east-1.amazonaws.com
Server:  UnKnown
Address:  192.168.0.1
*** UnKnown can't find https://c6lp4eoh2d-vpce-0c5db4e8da3c23739.execute-api.us-east-1.amazonaws.com: Non-existent domain

30 minutes later:
C:\Users>nslookup https://c6lp4eoh2d-vpce-0c5db4e8da3c23739.execute-api.us-east-1.amazonaws.com
Server:  UnKnown
Address:  192.168.0.1
*** UnKnown can't find https://c6lp4eoh2d-vpce-0c5db4e8da3c23739.execute-api.us-east-1.amazonaws.com: Non-existent domain

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Terraform never errors during the Terraform Apply - result is DNS never registered even after 30+ minutes 
### References

* #0000

**Comment by AdamTylerLynch** [comment]
I’ll be picking this one up.

**Comment by AdamTylerLynch** [comment]
@john-odonnell-accenture this issue has been resolved and will be released this week in v4.25.0
